src/controller.c

```diff
@@ -1,5 +1,79 @@
 #include "controller.h"
 
+// Gets rid of 'spawn()' implicit function declaration. Is this bad?
+#include "block_factory.h" 
+
+/*
+ * Modify stdin, check for valid user input, and revert stdin to original state.
+ *
+ * termios - http://man7.org/linux/man-pages/man3/tcsetattr.3.html
+ * fcntl - http://man7.org/linux/man-pages/man2/fcntl.2.html
+*/
+int is_user_input() {
+    struct termios old_term, new_term;
+    int old_fd;
+    int user_input;
+
+    // Modify stdin with saved copies of original state
+    tcgetattr(STDIN_FILENO, &old_term);
+    new_term = old_term;
+    new_term.c_lflag &= ~(ICANON | ECHO); // Enable canonical mode, echo input
+    tcsetattr(STDIN_FILENO, TCSANOW, &new_term);
+    old_fd = fcntl(STDIN_FILENO, F_GETFL, 0); // Copy stdin fd
+    fcntl(STDIN_FILENO, F_SETFL, old_fd | O_NONBLOCK); // Non-blocking
+
+    // Grab user input
+    user_input = getchar();
+
+    // Revert stdin to original state
+    tcsetattr(STDIN_FILENO, TCSANOW, &old_term);
+    fcntl(STDIN_FILENO, F_SETFL, old_fd);
+
+    // If user input is valid
+    if(user_input != EOF)
+    {
```

Comment:
can you move brace to the same line for consistency. it was one thing i meant to bring up along with the spaces/tabs thing but i forgot, and we kinda settled on the same line thing